pkg/slayers/path/scion/raw.go

```diff
@@ -530,34 +530,100 @@ func (s *Raw) GetCurrentHopField( /*@ ghost ubuf []byte @*/ ) (res path.HopField
 
 // SetHopField updates the HopField at a given index.
 // @ requires  0 <= idx
-// @ preserves acc(s.Mem(ubuf), R20)
-// @ preserves sl.Bytes(ubuf, 0, len(ubuf))
-// @ ensures   r != nil ==> r.ErrorMem()
+// @ requires  acc(s.Mem(ubuf), R20)
+// @ requires  sl.Bytes(ubuf, 0, len(ubuf))
+// pres for IO:
+// @ requires validPktMetaHdr(ubuf)
+// @ requires s.GetBase(ubuf).EqAbsHeader(ubuf)
+// @ requires len(s.absPkt(ubuf).CurrSeg.Future) > 0
+// @ ensures  acc(s.Mem(ubuf), R20)
+// @ ensures  sl.Bytes(ubuf, 0, len(ubuf))
+// @ ensures  r != nil ==> r.ErrorMem()
+// posts for IO:
```

Comment:
I'm starting to feel like these "posts for IO" annotations are a bit misleading, mostly because they are not about the IO behaviour, but instead about functional properties of the function. It just so happens that the correctness of the IO behaviour is predicated on the functional correctness of the packet processing. @Dspil would you be ok with us dropping these annotations, or do you think that it will still be useful for you to have the comments here?